pkg/ipam/sync_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	ipamv1alpha1 "github.com/liqotech/liqo/apis/ipam/v1alpha1"
+	networkingv1beta1 "github.com/liqotech/liqo/apis/networking/v1beta1"
+)
+
+var _ = Describe("Sync routine tests", func() {
+	const (
+		syncFrequency = 3 * time.Second
+		testNamespace = "test"
+	)
+
+	var (
+		ctx               context.Context
+		fakeClientBuilder *fake.ClientBuilder
+		now               time.Time
+		expiredThreshold  time.Time
+		expired           time.Time
```

Comment:
Expired looks misleading to me... isn't it a threshold under which you are not considering changes in the local state since the k8s client can be unaligned with the actual k8s state?

And cache is misleading too; it is the internal (or local) state of the ipam, isn't it?